hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2707,6 +2728,57 @@ public ServiceInfoEx getServiceInfo() throws IOException {
     return new ServiceInfoEx(getServiceList(), caCertPem);
   }
 
+  @Override
+  public boolean bootstrap(List<OMNodeInfo> omNodeInfos) throws IOException {
+    LOG.error("OM bootstrap command should be received via the RPC server");
+    throw new UnsupportedOperationException("OM bootstrap command should be " +
+        "received via the RPC server");
+  }
+
+  /**
+   * Add a new OM Node to the HA cluster.
+   */
+  public void addNewOMNode(String newOMNodeId) {
+    OMNodeDetails newOMNodeDetails = OMNodeDetails.getOMNodeDetailsFromConf(
+        getConfiguration(), getOMServiceId(), newOMNodeId);
+    if (newOMNodeDetails == null) {
+      // Load new configuration object to read in new peer information
+      setConfiguration(new OzoneConfiguration());
```

Comment:
This is run on all OMs (old OMs also). OzoneManager#addNewOMNode() is called when Ratis sends notifyConfigurationChanged(). And since the old OMs are not necessarily restarted, we need to reload the configuration.